### ISSUE TITLE
HTTP Status 404 – Not Found on web ui

### ISSUE THREAD

**Issue by boweic** [linuxserver/docker-unifi-controller#100]
[linuxserverurl]: https://linuxserver.io
[![linuxserver.io](https://raw.githubusercontent.com/linuxserver/docker-templates/master/linuxserver.io/img/linuxserver_medium.png)][linuxserverurl]

------------------------------

## Expected Behavior
I want to log into web ui via https://ip:8443

## Current Behavior
webpage shows "HTTP Status 404 – Not Found"

## Steps to Reproduce
1.go to ip:8443

## Environment
**OS:** macOS Big Sur 11.1
**CPU architecture:** x86_64
**How docker service was installed:**official docker desktop for mac

## Command used to create docker container (run/create/compose/screenshot)
`---
version: "2.1"
services:
  unifi-controller:
    image: lscr.io/linuxserver/unifi-controller
    container_name: unifi-controller
    environment:
      - PUID=501
      - PGID=20
      - MEM_LIMIT=1024M #optional
      - MEM_STARTUP=1024M #optional
    volumes:
      - /Users/boweic/docker/unifi-controller/config:/config
    ports:
      - 3478:3478/udp
      - 10001:10001/udp
      - 8080:8080
      - 8443:8443
      - 1901:1900/udp #optional
      - 8843:8843 #optional
      - 8880:8880 #optional
      - 6789:6789 #optional
      - 5514:5514/udp #optional
    restart: unless-stopped`
## Docker logs
`      
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.

[s6-init] ensuring user provided files have correct perms...exited 0.

[fix-attrs.d] applying ownership & permissions fixes...

[fix-attrs.d] done.

[cont-init.d] executing container initialization scripts...

[cont-init.d] 01-envfile: executing... 

[cont-init.d] 01-envfile: exited 0.

[cont-init.d] 10-adduser: executing... 

-------------------------------------

          _         ()

         | |  ___   _    __

         | | / __| | |  /  \ 

         | | \__ \ | | | () |

         |_| |___/ |_|  \__/

Brought to you by linuxserver.io

-------------------------------------

To support LSIO projects visit:

https://www.linuxserver.io/donate/

-------------------------------------

GID/UID

-------------------------------------

User uid:    501

User gid:    20

-------------------------------------

[cont-init.d] 10-adduser: exited 0.

[cont-init.d] 20-config: executing... 

[cont-init.d] 20-config: exited 0.

[cont-init.d] 30-keygen: executing... 

[cont-init.d] 30-keygen: exited 0.

[cont-init.d] 90-custom-folders: executing... 

[cont-init.d] 90-custom-folders: exited 0.

[cont-init.d] 99-custom-scripts: executing... 

[custom-init] no custom files found exiting...

[cont-init.d] 99-custom-scripts: exited 0.

[cont-init.d] done.

[services.d] starting services

[services.d] done.`

**Comment by daballiemo** [comment]
Same here, following helped:

docker exec -ti unifi bash
chown unifi.unifi -R /usr/lib/unifi
exit
docker restart unifi

**Comment by j0nnymoe** [comment]
This could be something specific to running this on MacOS as most of us within the team run this and haven't encountered this issue.

**Comment by daballiemo** [comment]
I am running this on a Synology NAS/Docker, for access I use MacOS BIg Sur 11.6. Problem occurs with Safari/Chrome/Firefox

**Comment by j0nnymoe** [comment]
Perfect, I'll test on my Synology soon to see if I can replicate the issue, could you give me your compose you used please?

**Comment by j0nnymoe** [comment]
also model/cpu platform/dsm version of synology too

**Comment by daballiemo** [comment]
Find below the relevant part of my compose file.
My synology setup is DS918+/DSM 7.0-41890


`# As for the unifi controller ->
  unifi:
    image: linuxserver/unifi-controller
    container_name: unifi
    ports:
      - 3478:3478/udp
      - 10001:10001/udp
      - 8080:8080
      - 8081:8081
      - 8443:8443
      - 8843:8843
      - 8880:8880
#      - 6789:6789
    volumes:
      - /volume1/docker/unifi:/config
    environment:
      - PUID=1000
      - PGID=1000
      - MEM_LIMIT=1024M
      - TZ=Europe/Amsterdam      
    restart: unless-stopped`

**Comment by lepoarro** [comment]
Same here: running on UNRAID 6.9.2 (repo: linuxserver/unifi-controller:latest). Log doesn't show anything:
```
-------------------------------------
_ ()
| | ___ _ __
| | / __| | | / \
| | \__ \ | | | () |
|_| |___/ |_| \__/


Brought to you by linuxserver.io
-------------------------------------

To support LSIO projects visit:
https://www.linuxserver.io/donate/
-------------------------------------
GID/UID
-------------------------------------

User uid: 99
User gid: 100
-------------------------------------

[cont-init.d] 10-adduser: exited 0.
[cont-init.d] 20-config: executing...
[cont-init.d] 20-config: exited 0.
[cont-init.d] 30-keygen: executing...
[cont-init.d] 30-keygen: exited 0.
[cont-init.d] 90-custom-folders: executing...
[cont-init.d] 90-custom-folders: exited 0.
[cont-init.d] 99-custom-scripts: executing...
[custom-init] no custom files found exiting...
[cont-init.d] 99-custom-scripts: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
```
But running on UNRAID 6.10.0-rc2 it is working. Same log: 
```
-------------------------------------
_ ()
| | ___ _ __
| | / __| | | / \
| | \__ \ | | | () |
|_| |___/ |_| \__/


Brought to you by linuxserver.io
-------------------------------------

To support LSIO projects visit:
https://www.linuxserver.io/donate/
-------------------------------------
GID/UID
-------------------------------------

User uid: 99
User gid: 100
-------------------------------------

[cont-init.d] 10-adduser: exited 0.
[cont-init.d] 20-config: executing...
[cont-init.d] 20-config: exited 0.
[cont-init.d] 30-keygen: executing...
[cont-init.d] 30-keygen: exited 0.
[cont-init.d] 90-custom-folders: executing...
[cont-init.d] 90-custom-folders: exited 0.
[cont-init.d] 99-custom-scripts: executing...
[custom-init] no custom files found exiting...
[cont-init.d] 99-custom-scripts: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
```

**Comment by R0cketM0nster** [comment]
I am also having the same issue on the latest build, if i pull down and run commit version 39a19f468565a53d290c417021e59e23b0dd73d3 this is working fine, something has changed in the most recent commit to break this.
I have tried this on Ubuntu 20.04.3.

**Comment by redespace** [comment]
I am having the same issue. I am running Armbian 21.08.3 Buster with Linux 5.10.60-meson64.
Ran docker compose using Portainer:
`
---
version: "2.1"
networks:
  Home:
    external:
      name: MyLAN
services:
  unifi-controller:
    image: ghcr.io/linuxserver/unifi-controller
    container_name: unifi-controller
    environment:
      - PUID=1000
      - PGID=100
    volumes:
      - /sharedfolders/Sys_Backup/OMV/appdata/Unifi:/config
    ports:
      - 3478:3478/udp
      - 10001:10001/udp
      - 8080:8080
      - 8443:8443
      - 1900:1900/udp #optional
      - 8843:8843 #optional
      - 8880:8880 #optional
      - 6789:6789 #optional
      - 5514:5514/udp #optional
    networks:
      Home:
        ipv4_address: 192.168.1.250
    restart: unless-stopped
`

Logs:
`
-------------------------------------
          _         ()
         | |  ___   _    __
         | | / __| | |  /  \ 
         | | \__ \ | | | () |
         |_| |___/ |_|  \__/
Brought to you by linuxserver.io
-------------------------------------
To support LSIO projects visit:
https://www.linuxserver.io/donate/
-------------------------------------
GID/UID
-------------------------------------
User uid:    1000
User gid:    100
-------------------------------------
[cont-init.d] 10-adduser: exited 0.
[cont-init.d] 20-config: executing... 
[cont-init.d] 20-config: exited 0.
[cont-init.d] 30-keygen: executing... 
Warning:
The JKS keystore uses a proprietary format. It is recommended to migrate to PKCS12 which is an industry standard format using "keytool -importkeystore -srckeystore /config/data/keystore -destkeystore /config/data/keystore -deststoretype pkcs12".
[cont-init.d] 30-keygen: exited 0.
[cont-init.d] 90-custom-folders: executing... 
[cont-init.d] 90-custom-folders: exited 0.
[cont-init.d] 99-custom-scripts: executing... 
[custom-init] no custom files found exiting...
[cont-init.d] 99-custom-scripts: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
WARN Unable to load properties from '/usr/lib/unifi/data/system.properties' - /usr/lib/unifi/data/system.properties (No such file or directory)
`

**Comment by neoPix** [comment]
Same here on raspberry 3 B (ubuntu server).

**Comment by fazy** [comment]
I've just had this issue on MacOS with M1 running Lima/QEMU (`Linux lima-default 5.13.0-20-generic #20-Ubuntu SMP Fri Oct 15 14:24:24 UTC 2021 aarch64 aarch64 aarch64 GNU/Linux`).

Initially I was using the default architecture (ARM), but [emulating X86](https://github.com/lima-vm/lima/blob/master/docs/multi-arch.md) has the same issue. Launched like this:

```
lima nerdctl run -d \
  --platform=amd64 \
  --name=unifi-controller \
  -e PUID=1000 \
  -e PGID=1000 \
  -e MEM_LIMIT=1024M `#optional` \
  -e MEM_STARTUP=1024M `#optional` \
  -p 3478:3478/udp \
  -p 10001:10001/udp \
  -p 8080:8080 \
  -p 8443:8443 \
  -p 1900:1900/udp `#optional` \
  -p 8843:8843 `#optional` \
  -p 8880:8880 `#optional` \
  -p 6789:6789 `#optional` \
  -p 5514:5514/udp `#optional` \
  -v /tmp/lima/unifi:/config \
  --restart no \
  lscr.io/linuxserver/unifi-controller
```

**Comment by leadhouse** [comment]
same here Unraid Version: 6.9.2

**Comment by sgade** [comment]
Same issue here. Running on ODroid C2 with Ubuntu minimal.
Note that I am running on a custom-built x86 machine with the same docker-compose file and that works just fine.

**Comment by Andriesmenze** [comment]
Same Issue Here on a Raspberry pi 4 After upgrading raspbian to the 64bit kernel, updating docker from 18.09.9 to 20.10.12, and updating the container to the latest image.

**Comment by xthursdayx** [comment]
I've also just run into the same issue on Unraid 6.10.0-rc2.

**Comment by walmartshopper** [comment]
I just ran into this trying to run the container on an OpenWrt router (Nanopi R4S, arm64).  Tried chown -R unifi:unifi but it didn't help.

**Comment by sgade** [comment]
I looked into the various log files that are available and it turns out mongo db was using an old driver that was failing to be found using the new version. In the end, it was easiest for me to reset the db of that install. Works fine since then.

**Comment by daballiemo** [comment]
> I looked into the various log files that are available and it turns out mongo db was using an old driver that was failing to be found using the new version. In the end, it was easiest for me to reset the db of that install. Works fine since then.

I do not see how this can be related to the fact that the problem is solved by setting permissions (solution quoted in the beginning of the issue). I think you have another problem that has the same consequences for the UI.

**Comment by walmartshopper** [comment]
I tried switching to an older tagged release. It failed to start with some error about the initial heap being too small.  I switched back to the latest / untagged release and now everything works and I no longer get the 404 error.

**Comment by redespace** [comment]
> I looked into the various log files that are available and it turns out mongo db was using an old driver that was failing to be found using the new version. In the end, it was easiest for me to reset the db of that install. Works fine since then.

@sgade How did you fix it? indeed, I was reading the logs and it seems to be a problem with MongoDB. I tried what @daballiemo posted, but that did not help.. still getting 404 :(

Tried Exec-ing into the container and ran Mongo:
```
root@dddbd1f0ed74:/usr/lib/unifi# mongo
MongoDB shell version v3.6.8
connecting to: mongodb://127.0.0.1:27017
2022-01-11T10:51:20.854+0000 W NETWORK  [thread1] Failed to connect to 127.0.0.1:27017, in(checking socket for error after poll), reason: Connection refused
2022-01-11T10:51:20.854+0000 E QUERY    [thread1] Error: couldn't connect to server 127.0.0.1:27017, connection attempt failed :
connect@src/mongo/shell/mongo.js:257:13
@(connect):1:6
exception: connect failed
```
Not sure what username/pwd Unifi uses.

**Comment by sgade** [comment]
@redespace I deleted all data and started fresh. That's because I could not find any working resources on how to upgrade to the new driver. If you have the option, I would recommend booting the old version up beforehand and saving a backup from the controller so that the restore process should be quicker. I was not minding losing any data with the install, so this method was quickest for me.

Overall, as it turns out, it was not a configuration fault of this docker image but rather of the Unifi controller IMO.

**Comment by redespace** [comment]
@sgade I tried that already. I removed everything what I could think of: the container, made sure there are no volumes left under /lib/docker, deleted the working directory and files, removed the image, and ran the compose again.. it still is not working for me :(..

again, these are my container logs:

```
[cont-init.d] 10-adduser: exited 0.
[cont-init.d] 15-install: executing... 
*** installing unifi packages ***
Selecting previously unselected package unifi.
(Reading database ... 
(Reading database ... 5%
(Reading database ... 10%
(Reading database ... 15%
(Reading database ... 20%
(Reading database ... 25%
(Reading database ... 30%
(Reading database ... 35%
(Reading database ... 40%
(Reading database ... 45%
(Reading database ... 50%
(Reading database ... 55%
(Reading database ... 60%
(Reading database ... 65%
(Reading database ... 70%
(Reading database ... 75%
(Reading database ... 80%
(Reading database ... 85%
(Reading database ... 90%
(Reading database ... 95%
(Reading database ... 100%
(Reading database ... 8986 files and directories currently installed.)
Preparing to unpack /app/unifi.deb ...
Unpacking unifi (6.5.55-16678-1) ...
Setting up unifi (6.5.55-16678-1) ...
invoke-rc.d: could not determine current runlevel
invoke-rc.d: policy-rc.d denied execution of start.
[cont-init.d] 15-install: exited 0.
[cont-init.d] 19-deprecate: executing... 
[cont-init.d] 19-deprecate: exited 0.
[cont-init.d] 20-config: executing... 
[cont-init.d] 20-config: exited 0.
[cont-init.d] 30-keygen: executing... 

Warning:
The JKS keystore uses a proprietary format. It is recommended to migrate to PKCS12 which is an industry standard format using "keytool -importkeystore -srckeystore /config/data/keystore -destkeystore /config/data/keystore -deststoretype pkcs12".
[cont-init.d] 30-keygen: exited 0.
[cont-init.d] 90-custom-folders: executing... 
[cont-init.d] 90-custom-folders: exited 0.
[cont-init.d] 99-custom-scripts: executing... 
[custom-init] no custom files found exiting...
[cont-init.d] 99-custom-scripts: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
```

inside container Server logs:

```
[2022-01-11T11:40:45,304] <main> INFO  system - *** Running for the first time, creating identity ***
[2022-01-11T11:40:45,315] <main> INFO  system - UUID: 7202a6a2-cbe5-4cda-876f-2ea5f3776a45
[2022-01-11T11:40:45,332] <main> INFO  system - Reporter UUID: cc428956-9fad-42a0-81e5-b72d7114f278
[2022-01-11T11:40:46,425] <launcher> INFO  tomcat - Adding basic REST API support during the startup
[2022-01-11T11:40:47,625] <launcher> INFO  system - Tomcat startup took 2012ms
[2022-01-11T11:40:47,628] <launcher> INFO  db     - New installation, skipping migration
[2022-01-11T11:40:47,661] <launcher> INFO  tomcat - Adding basic REST API support during the startup
[2022-01-11T11:40:49,699] <launcher> INFO  system - ======================================================================
[2022-01-11T11:40:49,700] <launcher> INFO  system - UniFi 6.5.55 (build atag_6.5.55_16678 - release/release) is started
[2022-01-11T11:40:49,700] <launcher> INFO  system - ======================================================================
[2022-01-11T11:40:49,702] <launcher> INFO  system - BASE dir:/usr/lib/unifi
[2022-01-11T11:40:49,814] <launcher> INFO  system - Current System IP: 192.168.1.250
[2022-01-11T11:40:49,815] <launcher> INFO  system - Hostname: d955cbe4bacb
[2022-01-11T11:40:49,816] <launcher> INFO  system - ubic.env: prod
[2022-01-11T11:40:50,376] <launcher> INFO  db     - waiting for db connection...
[2022-01-11T11:40:51,877] <launcher> INFO  db     - Connecting to mongodb://localhost:27117
[2022-01-11T11:40:51,884] <db-server> INFO  db     - 2022-01-11T11:40:51.882+0000 I STORAGE  [main] Engine custom option: cache_size=256M
[2022-01-11T11:40:51,993] <db-server> INFO  db     - DbServer stopped
[2022-01-11T11:40:56,197] <db-server> INFO  db     - 2022-01-11T11:40:56.196+0000 I STORAGE  [main] Engine custom option: cache_size=256M
[2022-01-11T11:40:56,263] <db-server> INFO  db     - DbServer stopped
[2022-01-11T11:41:00,377] <db-server> INFO  db     - 2022-01-11T11:41:00.377+0000 I STORAGE  [main] Engine custom option: cache_size=256M
[2022-01-11T11:41:00,455] <db-server> INFO  db     - DbServer stopped
[2022-01-11T11:41:04,645] <db-server> INFO  db     - 2022-01-11T11:41:04.644+0000 I STORAGE  [main] Engine custom option: cache_size=256M
[2022-01-11T11:41:04,702] <db-server> INFO  db     - DbServer stopped
```

here is the mongo db log:
[mongod.log](https://github.com/linuxserver/docker-unifi-controller/files/7846149/mongod.log)

**Comment by sgade** [comment]
@redespace There's an assertion failure in the mongo logs, looks like a different error to mine.

**Comment by Capusjon** [comment]
> Same here, following helped:
> 
> docker exec -ti unifi bash chown unifi.unifi -R /usr/lib/unifi exit docker restart unifi

Thank you, this did the trick

**Comment by fraser73** [comment]
I am also getting this on a pi4, using a windows server to provide the persistent volume storage, created thusly:

`docker pull linuxserver/unifi-controller

docker volume create --driver local --opt type=cifs --opt "o=username=unifi_service_id,password=PASSWORD,domain=DOMAIN,uid=1000,gid=1000,vers=3.0" --opt device=//IPADDRESS/unifi-config unifi-config

docker run -d --name=frinkunifi -e PUID=1000 -e PGID=1000 -p 3478:3478/udp -p 10001:10001/udp -p 8080:8080 -p 8443:8443 -p 1900:1900/udp -p 8843:8843 -p 8880:8880 -p 6789:6789 -p 5514:5514/udp -v unifi-config:/config --restart always lscr.io/linuxserver/unifi-controller`

MongoDB doesn't seem to be able to start, it gives errors in the mongodb.log like:

2022-02-02T15:42:29.009+0000 [initandlisten] ERROR: listen(): bind() failed errno:1 Operation not permitted for socket: /usr/lib/unifi/run/mongodb-27117.sock

**Comment by fraser73** [comment]
> We don't recommend having the /config volume on a remote file system. Mongodb doesn't play well with it, so use a local file system for the /config volume.

I've separated out the local /config volumes to the local filesystem on the Pi, and put the /config/data/backup as a separate SMB3 hosted persistent volume, so it can write the backups elsewhere, this seems to be working fine now.

**Comment by redespace** [comment]
I don't think this issue should be closed. Both odroidxu4 and ODROID-N2+ are experiencing this issue. I have shared the logs in this issue, and it seems to point to MongoDB. Can someone please look into this issue again?
Thank you

**Comment by daballiemo** [comment]
I have opened issue 123 as advised on Discord.

My issue is NOT related to Mongo DB but due to an file owner issue, as was the original ticket.

 

Kr

 

Han

 

From: redespace ***@***.***>
Reply to: linuxserver/docker-unifi-controller ***@***.***>
Date: Friday, 4 February 2022 at 19:39
To: linuxserver/docker-unifi-controller ***@***.***>
Cc: Daballiemo ***@***.***>, Mention ***@***.***>
Subject: Re: [linuxserver/docker-unifi-controller] HTTP Status 404 – Not Found on web ui (Issue #100)

 

I don't think this issue should be closed. Both odroidxu4 and ODROID-N2+ are experiencing this issue. I have shared the logs in this issue, and it seems to point to MongoDB. Can someone please look into this issue again?
Thank you

**Comment by TheSpad** [comment]
This issue has become a dumping ground for a series of unrelated issues with similar symptoms, which is why it's been closed.

@redespace your issue is more likely related to mongodb and the hardware you're running - if it's ongoing please open a new issue and copy/add your information so we can take a look, though I don't believe any of the team have ODROID hardware to work with so there is a limit to what we can test.

**Comment by gizmocuz** [comment]
I had the same issue, I had to specify the dns servers in the docker-compose file , remove the config folder, and restart the docker

```
    dns:
       - 8.8.8.8
       - 9.9.9.9
```

Seems it is downloading something at startup, without dns/internet it fails and i got 404

**Comment by tracetechnical** [comment]
Same issue here running 6.5.55 under K3S on an ubuntu host

**Comment by gizmocuz** [comment]
Be sure to wait a few minutes... seems when the docker is just started also this error happens and after a few minutes and refreshing the browser, it seems solved

**Comment by walmartshopper** [comment]
Same here... I get the 404 at startup but if I wait a bit it starts working.
> Be sure to wait a few minutes... seems when the docker is just started also this error happens and after a few minutes and refreshing the browser, it seems solved

**Comment by gizmocuz** [comment]
It's a bit annoying if you use the docker once in a while just to do a configuration or check for firmware updates and not using it to monitor 24/7.
I hope someone genius could have a look at this and see what is causing this delay

**Comment by TheSpad** [comment]
You'll likely see a similar issue with a native install, the delay is between us starting the unifi services and it bringing up the web front end.

**Comment by j0nnymoe** [comment]
@gizmocuz if you only have 1 AP, then you can just use the mobile app to configure it and do changes as and when you need. If you have more than one, well the controller is designed to run 24/7 as it then provides features like zero hand off which requires the controller running.

**Comment by gizmocuz** [comment]
No I have multiple AP's, a big switch and gateway
Maybe you could add a message when the docker is started (when you issue docker-compose up (without -d)) telling to please wait till all services are started, and when all services are started a 'Ready for use!' message
That would probably help a lot of users who now have a question like 'why isn't it working (now)'

**Comment by tracetechnical** [comment]
> I would argue this isn't happening to a lot of users. Nearly all the team run the unifi controller across multiple platforms and we've not been able to replicate this error at all. 

That isn't a good reason to #wontfix.... quite a few of us are having this issue.

**Comment by tracetechnical** [comment]
For reference: Using this under Kubernetes with NFS storage causes a permanent breakage, in case anyone else ends up here for the same reason I did. Not being able to have persistent non-host-local storage in a k8s environment is a massive PiTA.

**Comment by tracetechnical** [comment]
@TheSpad Dumping ground or not, I don't believe the original issue was ever addressed? Unless I've missed that part of the thread?

**Comment by gizmocuz** [comment]
Yes, that is why I mentioned it so others that experience the same 404 problem and are landing in this thread might solve their issues by waiting longer.

So if the issue is MongoDB ... Maybe it might be worth looking at SQLite or MariaDB/MySQL ?

**Comment by j0nnymoe** [comment]
All SQL has the same issue with remote mounts.

Besides, Unifi controller only supports mongodb.

**Comment by redespace** [comment]
I would like to reiterate that not everyone has an issue with remote mounts for MongoDB. In my case, it's just failing and I cannot determine why. If anyone is interested into looking at it, you can find all logs from my post in Jan 11 or here ([mongod.log](https://github.com/linuxserver/docker-unifi-controller/files/7846149/mongod.log).

At this point I just accept that this will not work anymore (it used to work btw), and I am now saving some money to actually buy the DreamMachine and forget about running this in a container...
`
2022-01-11T11:40:56.197+0000 I CONTROL  [main] ***** SERVER RESTARTED *****
2022-01-11T11:40:56.211+0000 I CONTROL  [initandlisten] MongoDB starting : pid=508 port=27117 dbpath=/usr/lib/unifi/data/db 64-bit host=d955cbe4bacb
2022-01-11T11:40:56.211+0000 I CONTROL  [initandlisten] db version v3.6.8
2022-01-11T11:40:56.211+0000 I CONTROL  [initandlisten] git version: 8e540c0b6db93ce994cc548f000900bdc740f80a
2022-01-11T11:40:56.211+0000 I CONTROL  [initandlisten] OpenSSL version: OpenSSL 1.1.1f  31 Mar 2020
2022-01-11T11:40:56.211+0000 I CONTROL  [initandlisten] allocator: tcmalloc
2022-01-11T11:40:56.211+0000 I CONTROL  [initandlisten] modules: none
2022-01-11T11:40:56.211+0000 I CONTROL  [initandlisten] build environment:
2022-01-11T11:40:56.212+0000 I CONTROL  [initandlisten]     distarch: aarch64
2022-01-11T11:40:56.212+0000 I CONTROL  [initandlisten]     target_arch: aarch64
2022-01-11T11:40:56.212+0000 I CONTROL  [initandlisten] options: { net: { bindIp: "127.0.0.1", port: 27117, unixDomainSocket: { pathPrefix: "/usr/lib/unifi/run" } }, processManagement: { pidFilePath: "/usr/lib/unifi/run/mongod.pid" }, storage: { dbPath: "/usr/lib/unifi/data/db", wiredTiger: { engineConfig: { configString: "cache_size=256M" } } }, systemLog: { destination: "file", logAppend: true, logRotate: "reopen", path: "/usr/lib/unifi/logs/mongod.log" } }
2022-01-11T11:40:56.213+0000 I STORAGE  [initandlisten] wiredtiger_open config: create,cache_size=1336M,session_max=20000,eviction=(threads_min=4,threads_max=4),config_base=false,statistics=(fast),cache_cursors=false,compatibility=(release="3.0",require_max="3.0"),log=(enabled=true,archive=true,path=journal,compressor=snappy),file_manager=(close_idle_time=100000),statistics_log=(wait=0),verbose=(recovery_progress),cache_size=256M
2022-01-11T11:40:56.247+0000 E STORAGE  [initandlisten] WiredTiger error (17) [1641901256:247479][508:0xffff8d52d040], connection: /usr/lib/unifi/data/db/WiredTiger.wt: handle-open: open: File exists
2022-01-11T11:40:56.247+0000 I STORAGE  [initandlisten] WiredTiger message unexpected file WiredTiger.wt found, renamed to WiredTiger.wt.1
2022-01-11T11:40:56.248+0000 E STORAGE  [initandlisten] WiredTiger error (1) [1641901256:248105][508:0xffff8d52d040], connection: /usr/lib/unifi/data/db/WiredTiger.wt: handle-open: open: Operation not permitted
2022-01-11T11:40:56.254+0000 E -        [initandlisten] Assertion: 28595:1: Operation not permitted src/mongo/db/storage/wiredtiger/wiredtiger_kv_engine.cpp 421
2022-01-11T11:40:56.254+0000 I STORAGE  [initandlisten] exception in initAndListen: Location28595: 1: Operation not permitted, terminating
2022-01-11T11:40:56.254+0000 I NETWORK  [initandlisten] shutdown: going to close listening sockets...
2022-01-11T11:40:56.254+0000 I NETWORK  [initandlisten] removing socket file: /usr/lib/unifi/run/mongodb-27117.sock
2022-01-11T11:40:56.254+0000 I CONTROL  [initandlisten] now exiting
2022-01-11T11:40:56.255+0000 I CONTROL  [initandlisten] shutting down with code:100
`

**Comment by j0nnymoe** [comment]
> I would like to reiterate that not everyone has an issue with remote mounts for MongoDB. In my case, it's just failing and I cannot determine why. If anyone is interested into looking at it, you can find all logs from my post in Jan 11 or here ([mongod.log](https://github.com/linuxserver/docker-unifi-controller/files/7846149/mongod.log).
> 
> At this point I just accept that this will not work anymore (it used to work btw), and I am now saving some money to actually buy the DreamMachine and forget about running this in a container...
> 
> ```
> 2022-01-11T11:40:56.247+0000 E STORAGE  [initandlisten] WiredTiger error (17) [1641901256:247479][508:0xffff8d52d040], connection: /usr/lib/unifi/data/db/WiredTiger.wt: handle-open: open: File exists
> 2022-01-11T11:40:56.247+0000 I STORAGE  [initandlisten] WiredTiger message unexpected file WiredTiger.wt found, renamed to WiredTiger.wt.1
> 2022-01-11T11:40:56.248+0000 E STORAGE  [initandlisten] WiredTiger error (1) [1641901256:248105][508:0xffff8d52d040], connection: /usr/lib/unifi/data/db/WiredTiger.wt: handle-open: open: Operation not permitted
> 2022-01-11T11:40:56.254+0000 E -        [initandlisten] Assertion: 28595:1: Operation not permitted src/mongo/db/storage/wiredtiger/wiredtiger_kv_engine.cpp 421
> 2022-01-11T11:40:56.254+0000 I STORAGE  [initandlisten] exception in initAndListen: Location28595: 1: Operation not permitted, terminating
> 2022-01-11T11:40:56.254+0000 I NETWORK  [initandlisten] shutdown: going to close listening sockets...
> 2022-01-11T11:40:56.254+0000 I NETWORK  [initandlisten] removing socket file: /usr/lib/unifi/run/mongodb-27117.sock
> 2022-01-11T11:40:56.254+0000 I CONTROL  [initandlisten] now exiting
> 2022-01-11T11:40:56.255+0000 I CONTROL  [initandlisten] shutting down with code:100
> ```

I think your issue might be specific to armbian/kernel though, I'm not sure anyone within the team currently has armbian running.

I've got a Pine64 I can spin up to test but won't be for a couple days at least

**Comment by j0nnymoe** [comment]
As this thread is getting long and out of control. It's clear that different setup's here are getting the same issue for different reasons. 

For that reason, Please can users here please open individual tickets with all the information that's requested in the template.

**Comment by maxheld83** [comment]
YMMV, but I was able to isolate my problem to the (recently added) **"Enable VirtioFS accelerated directory sharing"** on Docker Desktop (macOS).

The problem arose, as I had updated Docker Desktop and enabled VirtioFS (which is otherwise awesome), and I fixed the problem by disabling it again.

This may be related to [this issue on Docker Desktop for mac](https://github.com/docker/for-mac/issues/6243), which mentions a permissions issue for DB usage inside bind mounts (which is what I'm using).

This also (loosely) jibes with this output:

```
chown: changing ownership of '/config/data/sedMSD2ID': No such file or directory
```

(lots of those)

and the always shutting down mongoDB (as per `mongod.log`).

Anyway, this isn't a real root cause fix, but for me disabling virtioFS fixed this right away.
